src/main/java/com/urswolfer/intellij/plugin/gerrit/git/GerritGitUtil.java

```diff
@@ -113,6 +113,11 @@
 
     public Optional<GitRepository> getRepositoryForGerritProject(Project project, String gerritProjectName) {
         final Iterable<GitRepository> repositoriesFromRoots = getRepositories(project);
+        for (GitRepository repository : repositoriesFromRoots) {
+            if (repository.getRoot().equals(project.getBaseDir())) {
+                return Optional.of(repository);
```

Comment:
Can you try to explain in which case this return will be triggered? I'm still not sure if this could break backwards compatibility with existing setups of this plugins.

Reply:
It prefers the case, where the `.git` folder exists in the project root, where the source code is checked out, but submodules  on a other place higher in the direcory tree.
This code would break backwards compatibility only, if a submodule is checked out into project root. But that is a very exotic setup in my opinion.

Reply:
Do you think we can combine this new logic with also remoteUrl == gerritProjectName (as used below)? I think that would a bit more safe.

Reply:
I've just looked a bit into the code and had one more idea: we could also pass the branch name to `getRepositoryForGerritProject` and take it into account when comparing (check if remote branch name is equal). Do you think this would also help?

Another question: does you change also work when you want to check out a change which belongs to a sub project?

Thanks for you help!